data_classes.py

```diff
@@ -0,0 +1,139 @@
+from dataclasses import dataclass, field
+from datetime import datetime
+import configs.constants as constants
+import json
+import logging
+import logging.config
+
+with open("configs/log_config.json", "r", encoding="UTF-8") as stream:
+    config = json.load(stream)
+logging.config.dictConfig(config)
+logger = logging.getLogger("standard")
+
+
+@dataclass
+class TokenSale:
+    """
+    Data of given sale transaction
+    """
+
+    sort_index: int = field(init=False)
+    avax_price_in_usd: float
+    raw_sale: dict
+    price_floor: float
+    last_sales: list
+
+    def __post_init__(self):
+        self.sort_index = self.last_sales[0]["timestamp"]
+        self.transactionID: str = self.raw_sale["id"]
+        self.contractId: str = self.raw_sale["collection"]
+        self.tokenID: int = self.raw_sale["tokenId"]
+        self.verified: str = self.raw_sale["verified"]
+        self.date: datetime = datetime.utcfromtimestamp(
+            int(self.last_sales[0]["timestamp"])
+        )
+        self.priceAVAX = round(int(self.last_sales[0]["price"]) * 10 ** (-18), 3)
+        self.priceUSD: float = round(float(self.avax_price_in_usd) * self.priceAVAX, 1)
+
+        self.isTakerAsk: bool = self.last_sales[0]["isTakerAsk"]
+
+        self.soldByFull = self.last_sales[0]["fromAddress"]
+        self.boughtByFull = self.last_sales[0]["toAddress"]
+        self.soldBy: str = self.soldByFull[:5] + "..." + self.soldByFull[-4:]
+        self.boughtBy: str = self.boughtByFull[:5] + "..." + self.boughtByFull[-4:]
+
+        if self.last_sales[0]["image"]:
+            self.imgLink: str = self.last_sales[0]["image"].replace(" ", "%20")

```

Comment:
are you trying to urlencode the string? we should do it properly if so (https://www.urlencoder.io/python/)